server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -1829,8 +1938,18 @@ private Set<DataSegment> announceHistoricalSegmentBatch(
       PreparedBatch preparedBatch = handle.prepareBatch(buildSqlToInsertSegments());
       for (List<DataSegment> partition : partitionedSegments) {
         for (DataSegment segment : partition) {
-          preparedBatch.add()
-              .bind("id", segment.getId().toString())
+          String segmentId = segment.getId().toString();
+          Long schemaId = null;
+          Long numRows = null;
+
+          if (schemaPresent && segmentSchemaMapping.getSegmentIdToMetadataMap().containsKey(segmentId)) {
+            Pair<Long, Long> schemaInfoPair = fetchSegmentSchemaInformation(segmentSchemaMapping, segmentId, fingerprintSchemaIdMap);
+            schemaId = schemaInfoPair.lhs;
+            numRows = schemaInfoPair.rhs;
+          }
+
+          PreparedBatchPart preparedBatchPart = preparedBatch.add()
```

Comment:
This part need not be changed from the original code.
We should always bind `numRows` and `schemaId`, null or not.
This way, we don't need two different SQLs.

Reply:
We need two different SQL's since the DB doesn't contain the additional columns if the feature is disabled.